pdc/apps/release/models.py

```diff
@@ -72,24 +72,23 @@ def __unicode__(self):
 
     @property
     def active(self):
-        releases = Release.objects.filter(active=True).filter(product_version__product=self)
-        return bool(releases)
+        return self.active_release_count > 0
 
     @property
     def product_version_count(self):
-        return ProductVersion.objects.filter(product=self).count()
+        return self.productversion_set.count()
 
     @property
     def active_product_version_count(self):
-        return ProductVersion.objects.filter(product=self).filter(release__active=True).distinct().count()
+        return sum(1 for pv in self.productversion_set.all() if pv.active)
 
     @property
     def release_count(self):
-        return Release.objects.filter(product_version__product=self).count()
+        return sum(pv.release_count for pv in self.productversion_set.all())
 
     @property
     def active_release_count(self):
-        return Release.objects.filter(product_version__product=self).filter(active=True).distinct().count()
+        return sum(pv.active_release_count for pv in self.productversion_set.all())
```

Comment:
Should we prefetch_related('release_set') for productversion_set to reduce DB query number?